src/Persistence/MassTransit.Azure.Cosmos/AzureCosmos/NewtonsoftJsonCosmosClientFactory.cs

```diff
@@ -51,7 +54,7 @@ Lazy<CosmosClient> CreateClient<T>()
         {
             return new Lazy<CosmosClient>(() =>
             {
-                var clientOptions = new CosmosClientOptions();
+                var clientOptions = _cosmosClientOptions?.Value ?? new CosmosClientOptions();
```

Comment:
I might be wrong, but you do not need null check here.

Reply:
`?.` propagates the null to `Value` if `_cosmosClientOptions` is null. Then `??` will take the right side operand (`new CosmosClientOptions`) if the left side operand is null

Reply:
as I'm saying, it is impossible to get null for `IOptions<T>`

Reply:
Oh! I misread your comment. 
I might be wrong, but if there is no configure for this options, we can get null:

![image](https://user-images.githubusercontent.com/130716409/232230672-aeeac8b9-070f-48fa-8ca3-db72ffba8f22.png)

Reply:
yes, but we have `.AddOptions()` call inside the main assembly. but you can easy verify with tests (basically existing tests should just work)

Reply:
Ok I agree with you. Do you want me to keep the null check?

Reply:
of course, I like less code 😂

Reply:
done